File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 7/2/2025, 11:54:03 PM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        ans = 0
        minn = prices[0]
        for i in range(len(prices)):
            ans = max(ans, prices[i] - minn)
            minn = min(minn, prices[i])
        return ans
        

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 7/2/2025, 11:54:07 PM
class Solution {
    public int maxProfit(int[] prices) {
        int min=prices[0];
       int  profit=0;
        for(int i=0;i<prices.length;i++)
        {
            if(min>prices[i])
            {
                min=prices[i];
            }
            profit=Math.max(profit,prices[i]-min);
        }
        return profit;
    }
}